src/secure_tunnel_watcher.py

```diff
@@ -53,6 +79,44 @@ def on_stream_event(self, event: IoTCoreMessage) -> None:
                     "thing-name": "not_needed_see_argv"
                 }""")
 
+            http_proxy_config = "http-proxy-config.conf"
+            http_proxy_content = {}
+
+            http_proxy = os.getenv('HTTP_PROXY', None)
+
+            if http_proxy != None: 
+                # removed http
+                http_proxy_env = http_proxy.replace("http://", "")
+                try:
+                    # extract all http proxy information
+                    http_proxy_user, http_proxy_password, http_proxy_host, http_proxy_port = extract_http_proxy_url(http_proxy_env)
+                    if http_proxy_host != None and http_proxy_port != None:
+                        http_proxy_content = {
+                            "http-proxy-host": http_proxy_host,
+                            "http-proxy-port": http_proxy_port,
+                        }
+                        if http_proxy_user != None and http_proxy_password != None:
+                            http_proxy_content['http-proxy-auth-method'] = "UserNameAndPassword"
+                            http_proxy_content['http-proxy-username'] = http_proxy_user
+                            http_proxy_content['http-proxy-password'] = http_proxy_password
+                        else:
+                            http_proxy_content['http-proxy-auth-method'] = "None"
+                        
+                        http_proxy_content["http-proxy-enabled"] = True
+                    else:
+                        http_proxy_content["http-proxy-enabled"] = False
+                    
+                except Exception as e:
+                    print('Not able to parse proxy config: {}'.format(http_proxy_env))
+                    print('error: {}'.format(e))
+                    http_proxy_content["http-proxy-enabled"] = False
+                    
+            else:
+                http_proxy_content["http-proxy-enabled"] = False
+                    
+            with open(http_proxy_config, "w") as f:
+                    f.write(json.dumps(http_proxy_content))
```

Comment:
You can directly `json.dump()` to the `f` file object without `write`.

Reply:
done